docs/ruleset_functions/get_BPF_building_area_types_and_zones.md

```diff
@@ -0,0 +1,84 @@
+	WORKSHOP: WAREHOUSE,  
+	NONE: NONE, 
+	}```
+
+- Create a dict to store results`building_area_types_with_total_area_and_zones_dict`: `building_area_types_with_total_area_and_zones_dict = {}`
+- For each building segment in RMD: `for building_segment in RMD.building.building_segments:`
+	- If the building_segment lighting_building_area_type is defined: `if building_segment.lighting_building_area_type != NULL and building_segment.lighting_building_area_type != NONE`
+		- Assign the BPF_BAT to the building_segment_BPF_BAT using the lighting_space_lookup: `building_segment_BPF_BAT = building_area_lookup[building_segment.lighting_building_area_type]`
+		- Assign "BUILDING_SEGMENT_LIGHTING" to classification_source: `classification_source = "BUILDING_SEGMENT_LIGHTING"`
+	- Else look at each zone, and then each space and determine the building_segment_BPF_BAT using the largest space.lighting_space_type: `else:`
+		- Create a dictionary to keep track of the space types and areas: `building_segment_space_types_areas_dict = {}`
+		- For each zone in the building segment: `for zone in building_segment.zones:`
+			- Create a dictionary for the zone BPF_BAT space types using the function get_zone_BPF_BAT: `zone_BPF_BAT_dict = get_zone_BPF_BAT(B_RMD, zone.id)`
+			- For each mapped building area type in the zone_BPF_BAT_dict: `for space_BPF_BAT in zone_BPF_BAT_dict:`
+				- If the BAT doesn't exist yet: Add this BAT to the building_segment_space_types_areas_dict with its area: `if space_BPF_BAT not in building_segment_space_types: building_segment_space_types_areas_dict[space_BPF_BAT] = zone_BPF_BAT_dict[space_BPF_BAT]`
+				- Else: Add the area: `building_segment_space_types_areas_dict[space_BPF_BAT] += zone_BPF_BAT_dict[space_BPF_BAT]`
+		- Get the BPF_BAT with the largest floor area from building_segment_space_types_areas_dict: `building_segment_BPF_BAT = max(building_segment_space_types_areas_dict, key=lambda k: building_segment_space_types_areas_dict[k])`
+		- Assign "SPACE_LIGHTING" to classification_source: `classification_source = "SPACE_LIGHTING"`
+	- At this point, the building_segment_BPF_BAT has been defined by one of the two approaches. If the BPF_BAT doesn't exist already: add a nested dict for this BPF_BAT to the building_area_types_with_total_area_and_zones_dict : `if building_segment_BPF_BAT not in building_area_types_with_total_area_and_zones_dict: building_area_types_with_total_area_and_zones_dict[building_segment_BPF_BAT] = {"ZONE_IDS":[], "AREA":0, "CLASSIFICATION_SOURCE": classification_source}`
+	- If the BPF_BAT does exist already, and the classification_source is "SPACE_LIGHTING", update the dictionary item classification_source to "SPACE_LIGHTING" so that the recorded value is set to the weakest method used to determine any of the project's BPF_BATs: `elif classification_source == "SPACE_LIGHTING": building_area_types_with_total_area_and_zones_dict[building_segment_BPF_BAT]["CLASSIFICATION_SOURCE"] = "SPACE_LIGHTING"`
```

Comment:
I think this should say "If the BPF_BAT does exist already, and the classification_source is "BUILDING_SEGMENT_LIGHTING" instead of If the BPF_BAT does exist already, and the classification_source is "SPACE_LIGHTING"

Reply:
Just not understanding why it is checking if the classifcation_source equals "Space Lighting" and then assigning the same thing if it does equal that.

Reply:
This is just making sure that if the `classification_source=SPACE_LIGHTING` for <ins>any</ins> of the building segments that share a particular BPF_BAT, that the `classification_source` for that item in the dictionary gets set to `SPACE_LIGHTING`. 

When the item is created the first time a BPF_BAT is encountered in the iteration, it could be created with a `classification_source=BUILDING_SEGMENT_LIGHTING`. Another building segment with the same BPF_BAT might have `classification_source=SPACE_LIGHTING` so we would overwrite the dictionary item value in that case. "SPACE_LIGHTING" might be inconsequentially overwritten with identical "SPACE_LIGHTING" if done this way, the actual implementation may differ.